src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java

```diff
@@ -249,6 +249,16 @@ private void checkCommErrors() throws ServerException {
 		if (lastCommError != null) {
 			throw lastCommError;
 		}
+		long lastEventSeenAt = this.binlogEventListener.getLastEventSeenAt();
+		long lastEventAge = System.currentTimeMillis() - lastEventSeenAt;
+		long maxAllowedEventAge = Math.round(this.heartbeatInterval * this.heartbeatIntervalAllowance);
+		if (lastEventAge > this.heartbeatInterval * this.heartbeatIntervalAllowance) {
+			throw new ServerException(
+					"Last binlog event seen " + lastEventAge + "ms ago, exceeding " + maxAllowedEventAge + "ms allowance " +
+					"(" + this.heartbeatInterval + " * " + this.heartbeatIntervalAllowance + ")",
+					HEARTBEAT_FAILURE_ERROR_CODE,
+					"HY000");
+		}
```

Comment:
put in a separate function plz.

Reply:
I think that we should make an effort to reconnect rather than simply crashing.  see the logic in `ensureReplicatorThread` -- I'm sure we can hook into that code path and trigger a reconnect.

Reply:
I'll give it a go, but I reserve the right to an "I Told You So" if this ends up detecting an issue but unable to reconnect because computers are hard 😉

Reply:
you can have an I-told-you-so right now, for all i care.  they're pretty cheap in this day and age.